File: app/db/seed.py
"""Seed demo organizations, users, and portfolio data."""

from __future__ import annotations

from sqlalchemy import select
from sqlalchemy.orm import Session

from app.auth.security import hash_password
from app.db.models import Organization, OrganizationType, PortfolioLink, User, UserRole


def seed_platform_data(db: Session) -> None:
    if db.scalar(select(Organization).limit(1)):
        return

    idbi = Organization(
        name="IDBI Bank — MSME Lending",
        org_type=OrganizationType.BANK,
        registration_id="BANK-IDBI-001",
    )
    db.add(idbi)
    db.flush()

    shree_ganesh = Organization(
        name="Shree Ganesh Auto Components Pvt Ltd",
        org_type=OrganizationType.MSME,
        registration_id="UDYAM-MH-12-0012345",
    )
    green_fab = Organization(
        name="GreenFab Textiles LLP",
        org_type=OrganizationType.MSME,
        registration_id="UDYAM-GJ-24-0098765",
    )
    db.add_all([shree_ganesh, green_fab])
    db.flush()

    bank_password = hash_password("IDBI@2026")
    msme_password = hash_password("MSME@2026")

    bank_users = [
        User(
            email="admin@idbi.bank.in",
            password_hash=bank_password,
            full_name="Priya Sharma",
            role=UserRole.BANK_ADMIN,
            organization_id=idbi.id,
        ),
        User(
            email="credit@idbi.bank.in",
            password_hash=bank_password,
            full_name="Amit Verma",
            role=UserRole.BANK_CREDIT,
            organization_id=idbi.id,
        ),
        User(
            email="risk@idbi.bank.in",
            password_hash=bank_password,
            full_name="Neha Kapoor",
            role=UserRole.BANK_RISK,
            organization_id=idbi.id,
        ),
        User(
            email="rm@idbi.bank.in",
            password_hash=bank_password,
            full_name="Rahul Mehta",
            role=UserRole.BANK_RM,
            organization_id=idbi.id,
        ),
    ]
    msme_users = [
        User(
            email="rajesh@shreeganesh.in",
            password_hash=msme_password,
            full_name="Rajesh Patil",
            role=UserRole.MSME_OWNER,
            organization_id=shree_ganesh.id,
            msme_id="msme-demo-001",
        ),
        User(
            email="accounts@shreeganesh.in",
            password_hash=msme_password,
            full_name="Sunita Patil",
            role=UserRole.MSME_VIEWER,
            organization_id=shree_ganesh.id,
            msme_id="msme-demo-001",
        ),
        User(
            email="founder@greenfab.in",
            password_hash=msme_password,
            full_name="Anita Desai",
            role=UserRole.MSME_OWNER,
            organization_id=green_fab.id,
            msme_id="msme-greenfab-002",
        ),
    ]
    db.add_all(bank_users + msme_users)

    portfolio = [
        PortfolioLink(
            bank_org_id=idbi.id,
            msme_id="msme-demo-001",
            business_name="Shree Ganesh Auto Components Pvt Ltd",
            sector="auto_components",
            gstin="27AABCS1234F1Z5",
            relationship_manager="Rahul Mehta",
            credit_limit_inr=15_000_000,
        ),
        PortfolioLink(
            bank_org_id=idbi.id,
            msme_id="msme-greenfab-002",
            business_name="GreenFab Textiles LLP",
            sector="textiles",
            gstin="24AABCG5678H1Z2",
            relationship_manager="Rahul Mehta",
            credit_limit_inr=8_000_000,
        ),
        PortfolioLink(
            bank_org_id=idbi.id,
            msme_id="msme-techparts-003",
            business_name="TechParts Engineering Works",
            sector="manufacturing",
            gstin="29AABCT9012K1Z8",
            relationship_manager="Amit Verma",
            credit_limit_inr=5_000_000,
        ),
    ]
    db.add_all(portfolio)
    db.commit()

File: server/src/db/seed.ts
import bcrypt from "bcryptjs";
import type Database from "better-sqlite3";

const BANK_PASSWORD = bcrypt.hashSync("IDBI@2026", 10);
const MSME_PASSWORD = bcrypt.hashSync("MSME@2026", 10);
const GOVT_PASSWORD = bcrypt.hashSync("GOVT@2026", 10);
const REG_PASSWORD = bcrypt.hashSync("REG@2026", 10);

export function seedDatabase(db: Database.Database): void {
  const existing = db.prepare("SELECT id FROM organizations LIMIT 1").get();
  if (existing) return;

  const insertOrg = db.prepare(
    "INSERT INTO organizations (name, org_type, registration_id) VALUES (?, ?, ?)"
  );

  const idbi = insertOrg.run("IDBI Bank — MSME Lending", "bank", "BANK-IDBI-001").lastInsertRowid as number;
  const shree = insertOrg.run("Shree Ganesh Auto Components Pvt Ltd", "msme", "UDYAM-MH-12-0012345").lastInsertRowid as number;
  const green = insertOrg.run("GreenFab Textiles LLP", "msme", "UDYAM-GJ-24-0098765").lastInsertRowid as number;
  const msmeMin = insertOrg.run("Ministry of MSME — Scheme Administration", "government", "GOVT-MSME-001").lastInsertRowid as number;
  const sidbi = insertOrg.run("SIDBI — Refinance & Development", "government", "GOVT-SIDBI-001").lastInsertRowid as number;
  const rbi = insertOrg.run("Reserve Bank of India — Supervision", "regulatory", "REG-RBI-001").lastInsertRowid as number;
  const gstn = insertOrg.run("GSTN — Tax Compliance Authority", "regulatory", "REG-GSTN-001").lastInsertRowid as number;

  const insertUser = db.prepare(
    "INSERT INTO users (email, password_hash, full_name, role, organization_id, msme_id) VALUES (?, ?, ?, ?, ?, ?)"
  );

  const bankUsers: [string, string, string, number, string | null][] = [
    ["admin@idbi.bank.in", "Priya Sharma", "bank_admin", idbi, null],
    ["credit@idbi.bank.in", "Amit Verma", "bank_credit", idbi, null],
    ["risk@idbi.bank.in", "Neha Kapoor", "bank_risk", idbi, null],
    ["rm@idbi.bank.in", "Rahul Mehta", "bank_rm", idbi, null],
  ];
  for (const [email, name, role, orgId, msmeId] of bankUsers) {
    insertUser.run(email, BANK_PASSWORD, name, role, orgId, msmeId);
  }

  const msmeUsers: [string, string, string, number, string][] = [
    ["rajesh@shreeganesh.in", "Rajesh Patil", "msme_owner", shree, "msme-demo-001"],
    ["accounts@shreeganesh.in", "Sunita Patil", "msme_viewer", shree, "msme-demo-001"],
    ["founder@greenfab.in", "Anita Desai", "msme_owner", green, "msme-greenfab-002"],
  ];
  for (const [email, name, role, orgId, msmeId] of msmeUsers) {
    insertUser.run(email, MSME_PASSWORD, name, role, orgId, msmeId);
  }

  const govtUsers: [string, string, string, number][] = [
    ["admin@msme.gov.in", "Dr. Kavita Rao", "govt_admin", msmeMin],
    ["schemes@msme.gov.in", "Vikram Singh", "govt_scheme_officer", msmeMin],
    ["officer@sidbi.in", "Meera Joshi", "govt_sidbi_officer", sidbi],
  ];
  for (const [email, name, role, orgId] of govtUsers) {
    insertUser.run(email, GOVT_PASSWORD, name, role, orgId, null);
  }

  const regUsers: [string, string, string, number][] = [
    ["supervisor@rbi.org.in", "Arjun Nair", "reg_rbi_supervisor", rbi],
    ["compliance@gstn.gov.in", "Deepa Krishnan", "reg_gstn_officer", gstn],
    ["filings@mca.gov.in", "Sanjay Reddy", "reg_mca_officer", rbi],
    ["nbfc@rbi.org.in", "Lakshmi Iyer", "reg_nbfc_reviewer", rbi],
  ];
  for (const [email, name, role, orgId] of regUsers) {
    insertUser.run(email, REG_PASSWORD, name, role, orgId, null);
  }

  const insertPortfolio = db.prepare(
    "INSERT INTO portfolio_links (bank_org_id, msme_id, business_name, sector, gstin, relationship_manager, credit_limit_inr) VALUES (?, ?, ?, ?, ?, ?, ?)"
  );
  insertPortfolio.run(idbi, "msme-demo-001", "Shree Ganesh Auto Components Pvt Ltd", "auto_components", "27AABCS1234F1Z5", "Rahul Mehta", 15000000);
  insertPortfolio.run(idbi, "msme-greenfab-002", "GreenFab Textiles LLP", "textiles", "24AABCG5678H1Z2", "Rahul Mehta", 8000000);
  insertPortfolio.run(idbi, "msme-techparts-003", "TechParts Engineering Works", "manufacturing", "29AABCT9012K1Z8", "Amit Verma", 5000000);
}
